server/src/main/java/org/apache/iotdb/db/engine/compaction/level/LevelCompactionTsFileManagement.java

```diff
@@ -439,19 +439,26 @@ private void forkTsFileList(
 
   @Override
   protected void merge(long timePartition) {
-    merge(forkedSequenceTsFileResources, true, timePartition, seqLevelNum,
+    // whether execute merge chunk in this task
+    boolean isMerge = merge(forkedSequenceTsFileResources, true, timePartition, seqLevelNum,
         seqFileNumInEachLevel);
-    if (enableUnseqCompaction && unseqLevelNum <= 1 && forkedUnSequenceTsFileResources.size() > 0) {
+    if (enableUnseqCompaction && unseqLevelNum <= 1
+        && forkedUnSequenceTsFileResources.get(0).size() > 0) {
+      isMerge = true;
       merge(isForceFullMerge, getTsFileList(true), forkedUnSequenceTsFileResources.get(0),
           Long.MAX_VALUE);
     } else {
-      merge(forkedUnSequenceTsFileResources, false, timePartition, unseqLevelNum,
-          unseqFileNumInEachLevel);
+      isMerge = merge(forkedUnSequenceTsFileResources, false, timePartition, unseqLevelNum,
+          unseqFileNumInEachLevel) || isMerge;
+    }
+    // if merge in this merge task, execute next merge task to check if need continuous merge
+    if (isMerge && enableContinuousCompaction) {
+      merge(timePartition);
```

Comment:
this might mean during a merge process, another merge can be processed... why this method is needed

Reply:
and this recursion might have no termination conditions...

Reply:
we want this effect because we want to start a new compaction task after one compaction task ends